lib/realtime/server.ex

```diff
@@ -0,0 +1,147 @@
+defmodule Realtime.Server do
+  @moduledoc """
+  Fetches live data from RTR, and forwards it to connected clients.
+  """
+
+  use GenServer
+
+  alias Gtfs.Route
+  require Logger
+
+  defmodule Opts do
+    @type t :: %__MODULE__{
+            url: String.t(),
+            poll_delay: integer
+          }
+    @enforce_keys [
+      :url,
+      :poll_delay
+    ]
+    defstruct [
+      :url,
+      :poll_delay
+    ]
+  end
+
+  @type subscriptions :: %{optional(pid()) => [Route.id()]}
+
+  @type vehicles :: %{optional(Route.id()) => [Realtime.Vehicle.t()]}
+
+  @type state :: %{
+          url: String.t(),
+          poll_delay: integer(),
+          subscriptions: subscriptions(),
+          vehicles_timestamp: integer() | nil,
+          vehicles: vehicles()
+        }
+
+  # Client functions
+
+  @spec start_link(Opts.t()) :: GenServer.on_start()
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @spec start_mocked(Opts.t()) :: pid()
+  def start_mocked(opts) do
+    {:ok, pid} = GenServer.start_link(__MODULE__, opts)
+    pid
+  end
+
+  @spec subscribe([Route.id()], GenServer.server()) :: vehicles()
+  def subscribe(route_ids, server \\ nil) do
+    server = server || __MODULE__
+    GenServer.call(server, {:subscribe, route_ids, self()})
+  end
+
+  # GenServer callbacks
+
+  @impl true
+  def init(opts) do
+    initial_state = %{
+      url: opts.url,
+      poll_delay: opts.poll_delay,
+      subscriptions: Map.new(),
```

Comment:
It may just be that I haven't used `Registry` before, but I think that the complexity of communicating with the registry outweighs the complexity of maintaining our own subscriptions. It seems like the things that make Registry useful for most cases (being wrapped in its own GenServer, and allowing more general pubsub access) don't really benefit us since we only have one producer and it's already a GenServer.

Reply:
It's pretty straightforward: the API replaced its homegrown version with only a few lines. https://github.com/mbta/api/blob/master/apps/events/lib/events.ex has the implementation.